chain/client/src/types.rs

```diff
@@ -284,11 +284,13 @@ pub struct Query {
     pub path: String,
     pub data: Vec<u8>,
     pub id: String,
+    /// Whether to only query results that are final
+    pub is_final: bool,
 }
 
 impl Query {
-    pub fn new(path: String, data: Vec<u8>) -> Self {
-        Query { path, data, id: generate_random_string(10) }
+    pub fn new(path: String, data: Vec<u8>, is_final: bool) -> Self {
```

Comment:
Let's use enums instead of booleans:

* http://blakesmith.me/2019/05/07/rust-patterns-enums-instead-of-booleans.html
* https://www.luu.io/posts/dont-use-booleans/

Reply:
I don't know it it could be applied as a general rule, but it makes sense to use enum in this case